amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/hold/AddToHoldsBulkV1Tests.java

```diff
@@ -0,0 +1,519 @@
+import org.alfresco.rest.rm.community.model.hold.HoldBulkOperation.HoldBulkOperationType;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkOperationEntry;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkStatus;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkStatus.Status;
+import org.alfresco.rest.rm.community.model.hold.HoldBulkStatusCollection;
+import org.alfresco.rest.rm.community.model.hold.HoldChild;
+import org.alfresco.rest.rm.community.model.hold.HoldChildEntry;
+import org.alfresco.rest.rm.community.model.user.UserRoles;
+import org.alfresco.rest.search.RestRequestQueryModel;
+import org.alfresco.rest.search.SearchRequest;
+import org.alfresco.rest.v0.service.RoleService;
+import org.alfresco.utility.constants.UserRole;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.UserModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * API tests for adding items to holds via the bulk process
+ */
+public class AddToHoldsBulkV1Tests extends BaseRMRestTest
+{
+    private static final String ACCESS_DENIED_ERROR_MESSAGE = "Access Denied.  You do not have the appropriate " +
+        "permissions to perform this operation.";
+    private static final int NUMBER_OF_FILES = 30;
+    private final List<FileModel> addedFiles = new ArrayList<>();
+    private final List<UserModel> users = new ArrayList<>();
+    private final List<Hold> holds = new ArrayList<>();
+    private Hold hold;
+    private Hold hold2;
+    private Hold hold3;
+    private FolderModel rootFolder;
+    private HoldBulkOperation holdBulkOperation;
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private ContentActions contentActions;
+
+    @BeforeClass(alwaysRun = true)
+    public void preconditionForAddContentToHold()
+    {
+        STEP("Create a hold.");
+        hold = getRestAPIFactory().getFilePlansAPI(getAdminUser()).createHold(
+            Hold.builder().name("HOLD" + generateTestPrefix(AddToHoldsV1Tests.class)).description(HOLD_DESCRIPTION)
+                .reason(HOLD_REASON).build(), FILE_PLAN_ALIAS);
+        holds.add(hold);
+
+        STEP("Create test files.");
+        testSite = dataSite.usingAdmin().createPublicRandomSite();
+
+        rootFolder = dataContent.usingAdmin().usingSite(testSite).createFolder();
+        FolderModel folder1 = dataContent.usingAdmin().usingResource(rootFolder).createFolder();
+        FolderModel folder2 = dataContent.usingAdmin().usingResource(folder1).createFolder();
+
+        // Add files to subfolders in the site
+        for (int i = 0; i < NUMBER_OF_FILES; i++)
+        {
+            FileModel documentHeld = dataContent.usingAdmin()
+                .usingResource(new Random().nextBoolean() ? folder1 : folder2)
```

Comment:
It would be better to add them alternately, otherwise we might end up with test failures based on the random distribution of files.

Reply:
I get the idea, but IMO it's not good. Instead we should pick fixed scenarios to test which cover the cases we think are interesting.

I've seen really hard to investigate issues in Search Services tests when we used the underlying Solr test framework. It turned out that the failures were due to this function hidden away in the framework:
https://github.com/fnp/pylucene/blob/master/lucene-java-3.5.0/lucene/src/test-framework/java/org/apache/lucene/util/LuceneTestCase.java#L734
It would have been much better to simply test both cases every run, rather than having the failure happen one in ten(?) runs.

Reply:
Thank you for the explanation. I changed it to add test files alternately.